invenio_records_resources/services/files/service.py

```diff
@@ -95,72 +95,84 @@ def update_file_metadata(self, identity, id_, file_key, data, uow=None):
+            )
+        except KeyError:
+            return None
 
     def read_file_metadata(self, identity, id_, file_key):
         """Read the metadata of a file."""
         record = self.get_record(id_, identity, "read_files")
 
-        self.run_components("read_file_metadata", identity, id_, file_key, record)
+        try:
+            self.run_components("read_file_metadata", identity, id_, file_key, record)
 
-        return self.file_result_item(
-            self,
-            identity,
-            record.files[file_key],
-            record,
-            links_tpl=self.file_links_item_tpl(id_),
-        )
+            return self.file_result_item(
+                self,
+                identity,
+                record.files[file_key],
+                record,
+                links_tpl=self.file_links_item_tpl(id_),
+            )
+        except KeyError:
+            return None
 
     @unit_of_work()
     def extract_file_metadata(self, identity, id_, file_key, uow=None):
         """Extract metadata from a file and update the file metadata file."""
         record = self.get_record(id_, identity, "create_files")
-        file_record = record.files[file_key]
-
-        self.run_components(
-            "extract_file_metadata",
-            identity,
-            id_,
-            file_key,
-            record,
-            file_record,
-            uow=uow,
-        )
-
-        uow.register(RecordCommitOp(file_record))
-
-        return self.file_result_item(
-            self,
-            identity,
-            record.files[file_key],
-            record,
-            links_tpl=self.file_links_item_tpl(id_),
-        )
+        try:
+            file_record = record.files[file_key]
+
+            self.run_components(
+                "extract_file_metadata",
+                identity,
+                id_,
+                file_key,
+                record,
+                file_record,
+                uow=uow,
+            )
+
+            uow.register(RecordCommitOp(file_record))
+
+            return self.file_result_item(
+                self,
+                identity,
+                record.files[file_key],
+                record,
+                links_tpl=self.file_links_item_tpl(id_),
+            )
+        except KeyError:
+            return None
 
     @unit_of_work()
     def commit_file(self, identity, id_, file_key, uow=None):
         """Commit a file upload."""
         record = self.get_record(id_, identity, "create_files")
 
-        self.run_components("commit_file", identity, id_, file_key, record, uow=uow)
+        try:
+            self.run_components("commit_file", identity, id_, file_key, record, uow=uow)
 
-        return self.file_result_item(
-            self,
-            identity,
-            record.files[file_key],
-            record,
-            links_tpl=self.file_links_item_tpl(id_),
-        )
+            return self.file_result_item(
```

Comment:
Shall we simply check here if:
```
file_key in record.files
```
and return None? Same below.

Reply:
I was contemplating the same but i thought that it could also help catch any other keyErrors that may be raised, or allow for future exceptions to be added. What do you think?

Reply:
I think I would go with what @ntarocco said because catching all KeyErrors is not always acceptable. I.e now you use the KeyError because you anticipate that the file is not there. Then, returning None instructs a special behaviour to be applied in the view/ui. Imagine now that there is another KeyError just because there is a bug somewhere. You will get the same behaviour while hiding the actual bug thus making the debugging process much harder...